### DISCUSSION TITLE
Trouble with aligning ex.Text with ex.Actor properly

### DISCUSSION THREAD

**Discussion by RyanGrieb** [excaliburjs/Excalibur#2466] [Q&A] [answered]
Hi,
Currently I have a Button class that extends Actor, and a text component inside the button class like so:

```
class Button extends ex.Actor {

constructor(...) { 
...
this.text = new ex.Text({...})
}

public onInitialize() {
this.graphics.show(this.defaultSprite)
this.graphics.show(this.text)
}

}
```

What steps should I take to align the text component properly inside my actor? I don't see anything in the docs that would allow me to change the ```Text``` location relative to the Actor.

This is what it looks like currently:
![image](https://user-images.githubusercontent.com/6068039/185815079-8d202eb5-349b-4683-a41f-7520c4849bd6.png)

**Comment by eonarheim** [comment, marked as answer]
Hi @RyanGrieb thanks for opening a discussion!

I can think of a couple things you can try to re-position the text graphic.

* `ex.Text` draws from the bottom left of the text glyphs by default on the `ex.BaseAlign.Alphabetic`, try setting it to `ex.BaseAlgin.Top`? 

```typescript
this.text.font.baseAlign = ex.BaseAlign.Top;
```

* Try specifying a graphics offset when you call `show()` to shift the graphics down

```typescript
this.graphics.show(this.text, {
  offset: ex.vec(0, this.text.localbounds.height) 
});
```

**Reply by RyanGrieb** [reply]
Setting `text.font.baseAlign = ex.BaseAlign.Top` worked. Thank you!
![Screenshot_2022-08-23_17-25-13](https://user-images.githubusercontent.com/6068039/186276938-8d5e892c-1601-4a41-96fa-7aee03d1f2ed.png)
